python/Ganga/Utility/execute.py

```diff
@@ -80,24 +78,24 @@ def output(data):
     if update_env:
         update_script, env_file_pipes = env_update_script()
         script += update_script
-    script += "\nimport sys; sys.exit(0)"
     return script, (fdread, fdwrite), env_file_pipes
 
 
-def __reader(pipes, output_ns, output_var):
+def __reader(pipes, output_ns, output_var, require_output):
     """ This function un-pickles a pickle from a file and return it as an element in a dictionary
     Args:
         pipes (tuple): This is a tuple containing the (read_pipe, write_pipe) from os.pipes containing the pickled object
         output_ns (dict): This is the dictionary we should put the un-pickled object
         output_var (str): This is the key we should use to determine where to put the object in the output_ns
+        require_output (bool): Should the reader give a warning if the pickle stream is not readable
     """
     os.close(pipes[1])
     with os.fdopen(pipes[0], 'rb') as read_file:
         try:
             output_ns.update({output_var: pickle.load(read_file)})
         except Exception as err:
-            logger.error("Err: %s" % str(err))
-            raise  # EOFError triggered if command killed with timeout
+            if require_output:
+                logger.error('Error getting output stream from command: %s', err)
```

Comment:
Because this is run in a separate thread and so the raised exception will go nowhere and be picked up be no-one.